src/thor/bidirectional_astar.cc

```diff
@@ -1041,6 +1069,10 @@ std::vector<std::vector<PathInfo>> BidirectionalAStar::FormPath(GraphReader& gra
       const BDEdgeLabel& edgelabel = edgelabels_reverse_[edgelabel_index];
       const DirectedEdge* opp_edge = nullptr;
       GraphId opp_edge_id = graphreader.GetOpposingEdgeId(edgelabel.edgeid(), opp_edge, tile);
+      if (opp_edge == nullptr) {
+        LOG_INFO("graphreader.GetOpposingEdgeId(edgelabel.edgeid(), opp_edge, tile) is nullptr");
+        continue;
```

Comment:
```suggestion
        throw std::runtime_error("BidirectionalAStar::FormPath failed, tile no longer available");
```

Reply:
@kevinkreiser 
We have an issue inside `for (auto edge_itr = path_begin; edge_itr != path_end; ++edge_itr, ++edge_index) {`
We can `continue` loop. And I think in case we skip first edge we can't use `is_first_edge` flag. And `std::prev` for iterator at all. 
Maybe I should make `throw` instead of `continue` inside this loop?
Or will be better to make `prev_edge_itr` variable and use it instead of `is_first_edge` flag and `std::prev(edge_itr)`?

Reply:
About `trip_shape.size() == 0`. It can happen if we skip all iterations here
```
for (auto edge_itr = path_begin; edge_itr != path_end; ++edge_itr, ++edge_index) {
  const GraphId& edge = edge_itr->edgeid;
  graphtile = graphreader.GetGraphTile(edge, graphtile);
  if (graphtile == nullptr) {
    continue;
  }
```